js/components/sidebar-menu.js

```diff
@@ -48,6 +57,58 @@ export default class SidebarMenu extends PureComponent {
     onOptionChange(name, !this.options[name])
   }
 
+  getStoredModelText (modelId) {
+    let link = window.location.href.split('?')[0] + '?modelId=' + modelId
+    return (
+      <div>
+        <p className='save-state-text'>
+          Model code:<br />
+          <textarea className='copy-text' id='model-code' value={modelId} readOnly />
+        </p>
+        <p className='save-state-text'>
+          Link to model:<br />
+          <textarea className='copy-text' id='model-link' value={link} readOnly />
+        </p>
+      </div>)
+  }
+
+  showSaveDialog () {
+    this.setState({
+      saveDialogVisible: true
+    })
+  }
+
+  hideSaveDialog () {
+    this.setState({
+      saveDialogVisible: false
+    })
+    this.handleChange('playing', true)
```

Comment:
What if the model is paused during saving? It'll be started automatically. Not a big issue, though.
I think an easy fix would be to pause and start model in Plates component. When "modelSaving" state prop is changed to true, we can save "playing" value (note that we don't have to use React state for that, something like `this.<nameOfYourProp` is enough and even better - no re-rendering) and then restore it when "modelSaving" state prop is changed to false. Probably somewhere in `componentDidUpdate`.

Reply:
Good catch! I'll take care of that